packages/pouchdb-core/src/changes.js

```diff
@@ -96,8 +96,10 @@ function Changes(db, opts, callback) {
 
 
   if (!db.taskqueue.isReady) {
-    db.taskqueue.addTask(function () {
-      if (self.isCancelled) {
+    db.taskqueue.addTask(function (failed) {
```

Comment:
we weren't actually handling the case where the constructor failed and then the user did `db.changes()`. this is tested now